### PR TITLE
Added refl and trans

### PR DESCRIPTION
**Pull request by Risto-Stevcev**

### CODE REVIEW

**Comment by paf31** [comment]
Not sure if you noticed, but these are already provided by the `Category` instance.

**Comment by Risto-Stevcev** [comment]
Yeah I noticed them earlier. Would you be opposed to just defining `id = refl` and `compose = trans`? I feel like they're easier to grok at first glance.
Or maybe to document that the id instance means `reflexivity` and the compose instance means `transitivity`

**Comment by paf31** [comment]
Yes, I'm fine with adding these, but we should do as you say and define `Category` in terms of them.

**Comment by paf31** [comment]
Thanks!

File: src/Data/Leibniz.purs [hunk 1]
```diff
@@ -34,10 +34,18 @@ runLeibniz :: forall f a b. a ~ b -> f a -> f b
...
-  id = Leibniz id
+  id = refl
+
+-- | Equality is reflexive.
+refl :: forall a. a ~ a
+refl = Leibniz id
+
+-- | Equality is transitive.
+trans :: forall a b c. a ~ b -> b ~ c -> a ~ c
+trans (Leibniz f) (Leibniz g) = Leibniz (g <<< f)
```
Comment (no-longer-on-githu-b) [inline src/Data/Leibniz.purs:48]: `unsafeCoerce` is faster and identical.

**Pull request merged**